src/dygraph-canvas.js

```diff
@@ -204,6 +204,12 @@ DygraphCanvasRenderer._drawSeries = function(e,
       point = arr[i];
     }
 
+    // RRDtool-like plot, all lines are strictly horizontal and vertical so avoid using subpixel coordinates
+    if (stepPlot && strokeWidth == 1) {
+      point.canvasx = Math.round(point.canvasx+0.5)-0.5;
```

Comment:
Thanks for the PR! My one concern here is that you're mutating the `point` object, which is surprising behavior for a method called `_drawSeries`. I'd feel more comfortable if you created local `canvasx` and `canvasy` variables and modified those.

Does this visual optimization also make sense for thicker strokeWidth?

Reply:
I recently realized that my graphs, although being visually similar to ```stepPlot``` are very different in the nature of underlying data. The data is not values which are valid from the time of introduction toward the next value or the end of the times (e.g. a central bank-imposed interest rate) but rather are differentials of a counter value (e.g. traffic counter) taken at the time. It affect almost all aspect of how those plots to be drawn (the line go to past, not to future), stacked and how the gaps in data are filled in the graphs. So it seems to end up in intruducing another graph type in additional to existing ```stepPlot```.